app/src/main/java/com/example/chatplease/groupchatactivity.java

```diff
@@ -130,8 +129,9 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
 
     private void SaveMessageInfoToDatabase()
     {
-        String message = usermessageinput.getText().toString();
-        String messageKey = GroupNameRef.push().getKey();
+       @Nullable String message = usermessageinput.getText().toString();
+       @Nullable String messageKey = GroupNameRef.push().getKey();
```

Comment:
If `usermessageinput` and `GroupNameRef` are `@Nullable`, as you annotated them at the beginning of the file, then this line can throw a `NullPointerException`. Either these 2 fields can't be null, or your code here will produce the exception. Please look into this.

Also, `usermessageinput.getText()` returns an `Editable` object. This means `toString()` will be called from `CharSequence.java`, because `Editable` implements `CharSequence`. Are you sure this `String` returned from `toString()` is `@Nullable`?

Reply:
Sorry for this...I noticed the problem and removed Nullable